oeps/oep-0002-bp-repo-metadata.rst

```diff
@@ -60,34 +69,86 @@ Each repo will include a file ``openedx.yaml``, with the following keys:
+    Code runs in production web servers and has publicly accessible views/APIs for end-users.
+
+    eg. credentials
+
+``frontend-app``:
+    The repository primarily contains code for a frontend application.
+
+    Code runs in production devices (such as web browsers and mobile devices) and is publicly accessible by end-users.
+
+    eg. frontend-app-profile, edx-app-ios
+
+``framework``:
+    The repository contains shared functionality that forms part of a common framework used by multiple services or apps in the platform.
+
+    Code runs in production, behind the scenes, (intended to be) globally used throughout the platform.
+
+    eg. edx-drf-extensions, edx-ace, frontend-platform, paragon
 
 ``backend-service``:
-    A service that is run as a part of openedx but is not really interacted with by the end-users.  eg. xqwatcher or ecommerce-worker
+    A service that is run as part of the platform but is not directly accessed by end-users.
+
+    Code runs in production backend servers, typically within the firewall, and provides APIs for access by public-facing ``webservices``.
+
+    eg. xqueue-watcher, ecommerce-worker
 
 ``backend-tooling``:
-    Scripts or configuration used in conjunction with services or in support of openedx.
+    The repository contains scripts or configuration used in conjunction with backend services or in support of openedx.
+
+    Code does *not* run in production.
+
+    eg. ecommerce-scripts, repo-tools, testeng-ci
+
+``library``:
+    The repository is an installable package that is reusable but doesn't run standalone. eg. a pip installable python package
+
+    Code runs in production, behind the scenes, locally used by a part of the platform.
+
+    eg. frontend-component-footer, edx-milestones, user-util
 
 ``xblock``:
-    Indicating that a repo contains an xblock or xblock related tooling.
+    The repository contains an xblock or xblock related tooling.
+
+    Code runs in LMS and Studio as part of course run content.
+
+    eg. xblock-review, staff_graded-xblock
+
+Ownership
+---------
+While the responsibilities of an owner are outside the scope of this OEP, a repository's metadata should provide sufficient information to identify and discover the repository's owner.
+
+Granularity of Code Ownership
+*****************************
+Rather than designating human owners to each and every repository, we designate owners to only high-level and business-aligned repositories as described in ``owner.team`` in Keys_. All other repositories are indirectly owned by owners of the high-level repositories. This indirect relationship is described in the ``owner.repo`` designation in Keys_. See `Ownership Rationale`_.
+
+Granularity of Human Owners
+***************************
+We choose to assign singular teams (i.e., edX "squads") as owners, rather than individuals or other organizational structures (such as working groups or edX "themes"). See `Ownership Rationale`_.
```

Comment:
Just a clarifying question: for many edX themes, a squad is the second-lowest level in the hierarchy, with a "team" or "pod" being the lowest level (for example, I am on the Programs theme, in the Masters/MM squad, on the "Grades and assignments" team/pod.  Do you mean "squad" in that same organizational sense here?

Reply:
@iloveagent57  Thanks for asking. Our goal is to assign ownership to relatively "healthy and stable" teams so they can absorb the cost of ownership (via rotating pager duties and upkeep) and become long-term technical stewards/architects of what they own - both the thinkers and the doers of what they own.  We consider a team size of 5-6 people as "healthy" and so whatever that translates to within your theme.  I can update this paragraph to clarify that.